### PR TITLE
Add BoardListBlock component and story

### PR DESCRIPTION
**Pull request by riazaia**
- add compound component to display a list of boards
- show each boards slug by default
- show a board's description when a slug is clicked

`// TODO: lots!`

### CODE REVIEW

File: src/blocks/BoardListBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { extractCompounds } from "utils/compound-utils";
+
+const Empty: React.FC<unknown> = ({ children }) => <p>{children}</p>;
+
+const Item = ({ slug, description, _selected, _onSelected }: ItemProps) => (
+  <>
+    <p onClick={() => _onSelected?.(slug)}>!{slug}</p>
```
Comment (essential-randomness) [inline src/blocks/BoardListBlock.tsx:8]: If you're adding an onClick event you want to make this a `button` for accessibility reasons, even if you end up styling it not to look like a button at all.

File: stories/Blocks/BoardListBlock.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    return (
+      <div style={{ width: "500px" }}>
+        <BoardListBlock
+          icon=""
+          title="Boards"
+          selectedBoardSlug={selectedBoard}
+          onSelectBoard={(slug) => {
+            action("select-board")(slug);
+            if (slug === selectedBoard) {
```
Comment (essential-randomness) [inline stories/Blocks/BoardListBlock.stories.tsx:85]: You probably want `setSelectedBoard(slug === selectedBoard ? "" : slug)` (I know it feels less readable to most beginners, but you'll get used to it I swear :P)

File: stories/Blocks/BoardListBlock.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    link: {
+      href: "#href",
+      onClick: action("hrefClick"),
+    },
+  },
+];
+
+export const Simple: Story = {
+  render: function Render() {
+    const [selectedBoard, setSelectedBoard] = React.useState("");
```
Comment (essential-randomness) [inline stories/Blocks/BoardListBlock.stories.tsx:75]: I'd use `null` instead of `""`. If TS yells at you, you can use the generic `useState<string>`

**Review by essential-randomness — CHANGES_REQUESTED** [review]

File: stories/Blocks/BoardListBlock.stories.tsx (hunk 2 above)
Comment (riazaia) [reply stories/Blocks/BoardListBlock.stories.tsx:85]: Done, but wondering if I should change the empty string for `null` here too?

**Review by essential-randomness — APPROVED** [review]

**Pull request merged**